railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -60,13 +64,19 @@ def parse(column_definition)
           new(name, type, index_type, attr_options)
         end
 
-        def valid_type?(type)
-          DEFAULT_TYPES.include?(type.to_s) ||
-            ActiveRecord::Base.connection.valid_type?(type)
+        def dangerous_name?(name)
+          ActiveRecord::Base.dangerous_attribute_method?(name)
+        end
+
+        def unknown_type?(type)
+          type &&
+            DEFAULT_TYPES.exclude?(type.to_s) &&
+            ! ActiveRecord::Base.connection.valid_type?(type)
```

Comment:
It hard to read the use of `exclude?` in multiple conditions to me.
Let's keep it as it is.

Reply:
Thanks @kamipo. I've reverted the refactoring.